src/commands/ImportCommand.js

```diff
@@ -87,16 +87,52 @@ export default class ImportCommand extends Command {
 
     async.series(this.commits.map((sha) => (done) => {
       progressBar.tick(sha);
+      // Get a list of files in this commit
+      const filesInPatch = this.externalExecSync(`git ls-tree --name-only -r ${sha}`)
+      .split("\n")
+      // Filter out empties
+      .filter(function (file) {return !!file; })
+      // Sort by longest first, to avoid the case where a file that is shorter matches
+      // a partial longer file
+      .sort(function (a, b) {
+        if (a.length < b.length) {
+          return 1;
+        }
+        if (a.length > b.length) {
+          return -1;
+        }
+        return 0;
+      })
+      // Map files to original and urlEncoded
+      .map(function (file) {
+        return {
+          encodedString: encodeURIComponent(file),
+          originalString: file
+        };
+      });
 
       // Create a patch file for this commit and prepend the target directory
       // to all affected files.  This moves the git history for the entire
       // external repository into the package subdirectory, commit by commit.
-      const patch = this.externalExecSync(`git format-patch -1 ${sha} --stdout`)
+      let patch = this.externalExecSync(`git format-patch -1 ${sha} --stdout`);
+
+      // Replace all file names with url encoded file names
+      patch = filesInPatch.reduce(function (partiallyReplacedPatch, file) {
+        return partiallyReplacedPatch.replace(new RegExp(file.originalString, "mg"), file.encodedString);
```

Comment:
Is there any danger here of regex meta characters in a filename messing up the match?

Or will the URL encoding take care of that?